modules/dxp/apps/portal-search-tuning/portal-search-tuning-blueprints-admin-web/src/main/resources/META-INF/resources/js/edit_blueprint/tabs/QueryBuilder.js

```diff
@@ -105,9 +105,12 @@ function QueryBuilder({
 							id={element.id}
 							indexFields={indexFields}
 							initialUIConfigurationValues={
-								initialSelectedElements[index]
-									? initialSelectedElements[index]
-											.uiConfigurationValues
+								initialSelectedElements.find(
+									(item) => item.id === element.id
+								)
+									? initialSelectedElements.find(
+											(item) => item.id === element.id
+									  ).uiConfigurationValues
```

Comment:
Before this return statement maybe we can define this as a variable first so it's less repetitive. 

```javascript
const initialSelectedElement = initialSelectedElements.find(
	(item) => item.id === element.id
);
```

Then the prop could be
```javascript
initialUIConfigurationValues={initialSelectedElement ? initialSelectedElement.uiConfigurationValues : undefined}
```

Reply:
Thanks, added!!